constructor/osxpkg.py

```diff
@@ -423,6 +425,79 @@ def pkgbuild_prepare_installation(info):
         shutil.rmtree(f"{pkg}.expanded")
 
 
+def create_plugins(pages: list = None, codesigner: CodeSign = None):
+    if not pages:
+        return
+    elif isinstance(pages, str):
+        pages = [pages]
+
+    fresh_dir(PLUGINS_DIR)
+    xcodebuild = None
+
+    for page in pages:
+        xcodeproj_dirs = [
+            file.resolve()
+            for file in Path(page).iterdir()
+            if file.suffix == ".xcodeproj"
+        ]
+        if xcodeproj_dirs:
+            if not xcodebuild:
+                xcodebuild = shutil.which("xcodebuild")
+                if not xcodebuild:
+                    raise RuntimeError(
+                        "Plugin directory contains an uncompiled project,"
+                        " but xcodebuild is not available."
+                    )
+                try:
+                    subprocess.run([xcodebuild, "--help"], check=True, capture_output=True)
+                except subprocess.CalledSubprocessError:
+                    raise RuntimeError(
+                        "Plugin directory contains an uncomplied project,"
+                        " but xcodebuild requires XCode to compile plugins."
+                    )
```

Comment:
This could be a utility function to reduce the size of the loop.

Also:

```suggestion
            if not xcodebuild:
                xcodebuild = shutil.which("xcodebuild")
                if not xcodebuild:
                    raise RuntimeError(
                        "Plugin directory contains an uncompiled project,"
                        " but xcodebuild is not available."
                    )
                try:
                    subprocess.run([xcodebuild, "--help"], check=True, capture_output=True)
                except subprocess.CalledSubprocessError:
                    raise RuntimeError(
                        "Plugin directory contains an uncompiled project, "
                        "but xcodebuild requires XCode to compile plugins."
                    )
```